Insert an accountability section

nitech-code-of-conduct-suggestions-3 opened the pull request:
Put together a plan for enforcement-accountability, as this can be a major issue with a hard, constitution-based reference. This would help limit concern that unconscious bias will affect how the CoC is applied.

We haven't had a "you may disagree but it says so in the CoC" or equivalent as effective discussion-enders before, but that's only valid when there's some accountability around when and how consistently that line is (implicitly/explicitly) used, and who is affected positively/negatively over time.

nitech-code-of-conduct-suggestions-3 committed:
Insert an accountability section

Put together a plan for enforcement-accountability, as this can be much harder with a non-discursive, constitution-based reference point. This would help limit concern that unconscious bias will affect how the CoC is applied.

We haven't had a "you may disagree but it says so in the CoC" or equivalent as effective discussion-enders before, but that's only valid when there's some accountability around when and how consistently that line is (implicitly/explicitly) used, and who is affected positively/negatively over time.

mauricerkelly commented:
I welcome the inclusion of an Accountability section that _lightly_ documents what admins will do in response to alleged Code of Conduct violations.

However, my feeling is that this section is overkill right now. In fact, to me it reads like an over-reaction to the introduction of a Code of Conduct.

MarkXA reviewed (changes requested):
Happy to include an accountability section. However, I'd take some convincing that a final decision beyond "a majority decision of the admins" is necessary. The suggested content would be more appropriate for industrial tribunals than light-touch moderation.

nitech-code-of-conduct-suggestions-3 commented:
The reason that we need a Code of Conduct is that a light touch approach to standards in general hasn't worked, for this community or others.

@MarkXA's point about slimming this down is fine - it's better to have a full example, then trim to make it simple to run, than the reverse. Will do so.

However, light accountability: either accountability is important and finding a couple of randomly selected non-admins isn't hard, or it isn't, and having a "light accountability" section is misleading, and only reduces scope for addressing issues when admins aren't seeing them, for any reason. It would be better to close the PR and leave it to community discussion, if the alternative is an accountability section that would have no meaningful accountability outside the admin (or admin-selected) team and that only stifles the discussion.

As to whether it's needed - if an admin is OK with responding to wording of an accountability section with an "anti-Code-of-Conduct reactionary" insinuation and [all](https://www.cio.com/article/3156565/developer/should-software-developers-have-a-code-of-ethics.html) [that](https://www.reddit.com/r/technology/comments/3fpnuw/githubs_new_code_of_conduct_says_our_open_source/) [implies](http://internet.gawker.com/the-gamergate-decision-shows-exactly-whats-broken-about-1682639327), then it is (links for illustration of why this is pejorative, not endorsement). @MarkXA 's response is reasonable, even if I disagree, but if we're expecting team-internal accountability, @mauricerkelly's previous comment should have been called out.

For clarity: this is no more a personal comment about the current admins (who have all contributed to the community massively, but are likely to rotate over time), than a Code of Conduct is about the current community (which we're seeking to protect and enhance).

Will provide a stripped down version (or please close the PR if consensus is it's unnecessary)

mauricerkelly commented:
@nitech-code-of-conduct-suggestions-3 

> The reason that we need a Code of Conduct _is that a light touch approach to standards in general hasn't worked, for this community_ or others.

This comment above suggests that there have been problems with how issues have been dealt with in the past. If you feel this way, please get in touch with an admin (or more than one admin) rather than through this PR.

andrewbolster reviewed (changes requested):
IMO this is massive overkill for what is effectively creating a new tier of administrative culpability. 

Would a better idea not to just rotate out one admin every quarter instead of putting together an inquisition team?

Also, the idea that this would be ~1hr a quarter is ludicrous.

nitech-code-of-conduct-suggestions-3 committed:
Simplify accountability section

nitech-code-of-conduct-suggestions-3 commented:
Pushed some simplifications, making it a few summary lines:

> To help ensure consistency and transparency of enforcement, two non-admins, picked at random from the community, will act as an accountability panel. Their job is to provide feedback on the enforcement of the Code of Conduct. When the Code of Conduct is invoked, they should be informed where possible by admins, without personal details, and members of the community can give feedback to them anonymously. Each quarter they provide a brief summary of feedback, if any, in a public channel and two new members are picked. They have no admin powers.

Reading through the discussion though:

> > The reason that we need a Code of Conduct is that a light touch approach to standards in general has not worked, for this community or others.
>
> This comment above suggests that there have been problems with how issues have been dealt with in the past. If you feel this way, please get in touch with an admin (or more than one admin) rather than through this PR.

@mauricerkelly has done an excellent job, which is obvious from talking to anyone in the Slack, but I still do not think that is a working substitute for a Code of Conduct - @mauricerkelly has pointed out this out too. _Light touch_ does not provide a defined structure to build trust on, or to help people know on arrival what they can expect and what is expected of them. And, as dedicated as an admin can be, if the rest of the community are not clear on when to flag things up, admins cannot be in every channel all the time to spot issues, and people will overstep without realising, in NIT&D Slack as anywhere else. A Code of Conduct makes this transparent, helps offenders understand they are not being singled out, and makes it clear that admins are enforcing a clear set of community-agreed standards. So, a working approach involves a Code of Conduct.

In terms of time: knowing who is involved, this is more about demonstrating a working appeals mechanism than expecting anyone to have to use it - there should be virtually no work needing to be done under this for panelists (other than receiving admin notifications), no formal meetings and for writing a quick five line summary ("report"), 2-3 person hours / quarter, on average, should be plenty. Still, any system generally creates much more work than planned, so fair criticism. Rotating admins too regularly makes it likely to get someone less effective, or sensitive, than the admins who have been carefully chosen and understand what is needed - that is one reason it would be helpful to have something separate.

This was an idea to see if we could go to the next step, have external accountability as part of transparency, as in other parallel processes outside the IT community. This PR discussion was intended to refine that idea, but looking at the responses (and +1), I think there is consensus that this is not the right way to get there, so I will close this PR. If someone else wants to provide an alternative, I think a fresh start would be better.
